src/flows/pipes/Notification/view.tsx

```diff
@@ -259,243 +255,21 @@ const Notification: FC<PipeProp> = ({Context}) => {
-    ).reduce((acc, curr) => {
-      // eslint-disable-next-line no-extra-semi
-      ;(curr.assignment?.init?.properties || []).reduce((_acc, _curr) => {
-        if (_curr.key?.name && _curr.value?.location?.source) {
-          _acc[_curr.key.name] = _curr.value.location.source
-        }
-
-        return _acc
-      }, acc)
-
-      return acc
-    }, {})
-
-    vars.timeRangeStart = `-${deadman?.deadmanStopValue}`
-
-    const params = remove(
-      ast,
-      node =>
-        node.type === 'OptionStatement' && node.assignment.id.name === 'task'
-    ).reduce((acc, curr) => {
-      // eslint-disable-next-line no-extra-semi
-      ;(curr.assignment?.init?.properties || []).reduce((_acc, _curr) => {
-        if (_curr.key?.name && _curr.value?.location?.source) {
-          _acc[_curr.key.name] = _curr.value.location.source
-        }
-
-        return _acc
-      }, acc)
-
-      return acc
-    }, {})
-
-    const conditions = THRESHOLD_TYPES[deadmanType].condition(deadman)
-
-    const newQuery = `
-import "strings"
-import "regexp"
-import "influxdata/influxdb/monitor"
-import "influxdata/influxdb/schema"
-import "influxdata/influxdb/secrets"
-import "experimental"
-${ENDPOINT_DEFINITIONS[data.endpoint]?.generateImports()}
-
-check = {
-	_check_id: "${id}",
-	_check_name: "Notebook Generated Deadman Check",
-	_type: "deadman",
-	tags: {},
-}
-
-notification = {
-	_notification_rule_id: "${id}",
-	_notification_rule_name: "Notebook Generated Rule",
-	_notification_endpoint_id: "${id}",
-	_notification_endpoint_name: "Notebook Generated Endpoint",
-}
-
-task_data = ${format_from_js_file(ast)}
-trigger = ${conditions}
-messageFn = (r) => ("${data.message}")
-
-${ENDPOINT_DEFINITIONS[data.endpoint]?.generateQuery(data.endpointData)}
-|> monitor["deadman"](t: experimental["subDuration"](from: now(), d: ${
-      deadman.deadmanCheckValue
-    }))`
-
-    const newAST = parse(newQuery)
-
-    if (!params.name) {
-      params.name = `"Notebook Deadman Task for ${id}"`
-    }
-
-    if (data.interval) {
-      params.every = data.interval
-    }
-
-    if (data.offset) {
-      params.offset = data.offset
-    }
-
-    if (Object.keys(vars).length) {
-      const varString = Object.entries(vars)
-        .map(([key, val]) => `${key}: ${val}`)
-        .join(',\n')
-      const varHeader = parse(`option v = {${varString}}\n`)
-      newAST.body.unshift(varHeader.body[0])
-    }
-
-    const paramString = Object.entries(params)
-      .map(([key, val]) => `${key}: ${val}`)
-      .join(',\n')
-    const taskHeader = parse(`option task = {${paramString}}\n`)
-    newAST.body.unshift(taskHeader.body[0])
-
-    return format_from_js_file(newAST)
-  }, [
-    id,
-    queryText,
-    data.every,
-    data.offset,
-    data.endpointData,
-    data.endpoint,
-    data.thresholds,
-    data.message,
-  ])
-
-  const generateTask = useCallback(() => {
-    if (data.thresholds[0].type === deadmanType) {
```

Comment:
note: moving this was one change

Reply:
i missed where you reimplemented this check between the threshold types

Reply:
The same logic is not there.

Only difference is that the root method generateTask() now does the hand off to the helper methods.
```
  const generateTask = useCallback(() => {
    event('Alert Panel (Notebooks) - Export Alert Task Clicked')

    if (data.thresholds[0].type === deadmanType) {
      return generateDeadmanTask()
    } else {
      return generateThresholdTask()
    }
  }, [generateDeadmanTask, generateThresholdTask, data.thresholds])
```

Reply:
https://github.com/influxdata/ui/pull/3916/files#diff-8061db6ad62c27e1ef312918e57f5cfe61c64cfab70532e30477c0d8314ae72bR265  there it is